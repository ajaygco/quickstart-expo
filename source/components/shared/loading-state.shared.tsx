// Modules
import * as React from "react";
import { ActivityIndicator, View } from "react-native";
import type { ActivityIndicatorProps } from "react-native";

// Component
export const LoadingState: React.FC<ActivityIndicatorProps> = (
  props,
): React.ReactElement => {
  return (
    <View className="items-center justify-center">
      <ActivityIndicator {...props} />
    </View>
  );
};
