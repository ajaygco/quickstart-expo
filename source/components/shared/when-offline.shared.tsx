// Modules
import * as React from "react";
import { useIsConnected } from "react-native-offline";

// Types
interface WhenOfflineProps {
  children: React.ReactElement;
}

// Component
export const WhenOffline = ({
  children,
}: WhenOfflineProps): React.ReactElement | null => {
  // Hooks
  const isDeviceConnected: boolean | null = useIsConnected();

  if (isDeviceConnected) {
    return null;
  }

  return children;
};
