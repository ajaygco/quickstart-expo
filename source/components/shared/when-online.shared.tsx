// Modules
import * as React from "react";
import { useIsConnected } from "react-native-offline";

// Types
interface WhenOnlineCommonProps {
  children: React.ReactElement;
}
export interface WhenOnlineUiProps extends WhenOnlineCommonProps {
  isConnected: boolean | null;
}
export interface WhenOnlineProps extends WhenOnlineCommonProps {}

// Component: Presentation
export const WhenOnlineUi = ({
  children,
  isConnected,
}: WhenOnlineUiProps): React.ReactElement | null => {
  if (isConnected) {
    return children;
  }

  return null;
};

// Component: Logic
export const WhenOnline = (props: WhenOnlineProps): React.ReactElement => {
  // Hooks
  const isDeviceConnected: boolean | null = useIsConnected();

  return <WhenOnlineUi isConnected={isDeviceConnected} {...props} />;
};
