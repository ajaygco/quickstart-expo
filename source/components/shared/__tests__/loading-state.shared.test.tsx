// bun run test source/components/shared/__tests__/loading-state.shared.test.tsx

import * as React from "react";
import { render } from "@testing-library/react-native";

import { LoadingState } from "@app/components/shared/loading-state.shared";

describe("LoadingState", () => {
  it("should render correctly", () => {
    const wrapper = render(<LoadingState />);

    expect(wrapper).toMatchSnapshot();
  });
});
