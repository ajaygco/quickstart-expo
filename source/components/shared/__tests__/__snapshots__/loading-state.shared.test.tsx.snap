// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingState should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator />
</View>
`;
